Krish-Bhalala/Acadmeic-Learning-Centre-Admin-Scripts
# Feature/email sending feature

**What does this PR do?**
This PR extends the existing timesheet automation script by adding a dedicated email sending feature. It introduces a new `sendEmail` function, which leverages SMTP to dispatch the generated email data from `emailDataGenerator`. With this enhancement, once the timesheet is processed, the updated sheet can be emailed out automatically, streamlining the entire workflow.

**Key Changes:**
- Created a `sendEmail` function that uses `smtplib` and `email.message.EmailMessage` to send emails.
- Integrated `emailDataGenerator` and `sendEmail` into the timesheet logic.
- Added robust error handling to ensure that failed sends return a clear message.

**Why is this needed?**
Previously, the script only prepared the email data. Now, the addition of the `sendEmail` function allows for a fully automated process: after updating timesheets, the script can immediately send the resulting information to the intended recipients. This reduces manual overhead and keeps stakeholders updated efficiently.

**Testing Steps:**
1. Update the `directory_path`, `file_path`, `TUTOR_ID`, `EMPLOYEE_ENDPOINT`, and SMTP credentials.
2. Run the timesheet script as before.
3. Confirm that the timesheet is processed successfully.
4. Check whether an email is dispatched to the recipient, verifying that the subject, body, and sender details are correct.

**Additional Notes:**
- Ensure that the SMTP credentials and any application-specific passwords are securely managed.
- If using Gmail, make sure to enable necessary settings or use an App Password for the SMTP connection.
fixes #3
